Network/Socket.hs

```diff
@@ -100,6 +100,44 @@
 -- unexpected things would happen. There is one exception for multiple
```

Comment:
This existing sentence is not good.

`unexpected things`, what's that referring to?

This needs explanation.